src/lib/AutoControlledComponent.js

```diff
@@ -183,8 +183,9 @@ export default class AutoControlledComponent extends Component {
    * Second argument is a state object that is always passed to setState.
    * @param {object} maybeState State that corresponds to controlled props.
    * @param {object} [state] Actual state, useful when you also need to setState.
+   * @param {function} [callback] a setState callback
    */
-  trySetState = (maybeState, state) => {
+  trySetState = (maybeState, state, callback = () => {}) => {
```

Comment:
Thanks for this, I'd been wanting to update this to match React's API.  There are only a few cases where we use the `state` argument as well.  I'm thinking we should also remove it to match the React API 1:1.  Otherwise, to use the callback you need to pass some null `state`:
```jsx
this.trySetState(
  { count: 1 },             // maybeState
  null,                     // state, I don't need...
  () => console.log('done') // complete callback
)
```

The current `trySetState` implementation throws warnings about including state keys that are not listed in `autoControlledProps`, however, it could instead ignore them and include them as-is in the actual `setState` call.  This way we can just combine the maybe state and the actual state.

Reply:
I tried to remove the `state` arguments out of  `trySetState`.
I'm not sure I did understand what you said.
please review this code again